packages/cli/generators/model/index.js

```diff
@@ -0,0 +1,273 @@
+        )}:`,
+        type: 'list',
+        default: baseOptions[0],
+        choices: baseOptions,
+        when: baseOptions.length > 1,
+      },
+    ];
+
+    return this.prompt(prompts).then(props => {
+      if (!props.base) props.base = baseOptions[0];
+      Object.assign(this.artifactInfo, props);
+
+      this.log(
+        `Let's add a property to ${chalk.yellow(this.artifactInfo.className)}`,
+      );
+
+      return props;
+    });
+  }
+
+  // Prompt for a property name
+  promptPropertyName() {
+    this.log(`Enter an empty property name when done`);
+
+    delete this.propName;
+    const prompts = [
+      {
+        name: 'propName',
+        message: 'Enter the property name:',
+        validate: function(val) {
+          if (val) {
+            return utils.checkPropertyName(val);
+          } else {
+            return true;
+          }
+        },
+      },
+    ];
+
+    return this.prompt(prompts).then(props => {
+      if (props.propName) {
+        this.artifactInfo.properties[props.propName] = {};
+        this.propName = props.propName;
+      }
+      return this._promptPropertyInfo();
+    });
+  }
+
+  // Internal Method. Called when a new property is entered.
+  // Prompts the user for more information about the property to be added.
+  _promptPropertyInfo() {
+    if (!this.propName) {
+      return true;
+    } else {
+      const prompts = [
+        {
+          name: 'type',
+          message: 'Property type:',
+          type: 'list',
+          choices: this.typeChoices,
+        },
+        {
+          name: 'arrayType',
+          message: 'Type of array items:',
+          type: 'list',
+          choices: this.typeChoices.filter(choice => {
+            return choice !== 'array';
+          }),
+          when: function(answers) {
+            return answers.type === 'array';
+          },
+        },
+        {
+          name: 'id',
+          message: 'Is ID field?',
+          type: 'confirm',
+          default: false,
+          when: function(answers) {
+            return (
+              !this.idFieldSet &&
+              !['array', 'object', 'buffer'].includes(answers.type)
+            );
+          }.bind(this),
+        },
+        {
+          name: 'required',
+          message: 'Required?:',
+          type: 'confirm',
+          default: false,
+        },
+        {
+          name: 'default',
+          message: `Default value ${chalk.yellow('[leave blank for none]')}:`,
+          when: function(answers) {
+            return (
+              ![null, 'buffer', 'any'].includes(answers.type) &&
+              this.typeChoices.includes(answers.type)
+            );
+          }.bind(this),
+        },
+      ];
+
+      return this.prompt(prompts).then(props => {
+        if (props.default === '') {
+          delete props.default;
+        }
+
+        Object.assign(this.artifactInfo.properties[this.propName], props);
+        if (props.id) {
+          this.idFieldSet = true;
```

Comment:
I love the check here to prevent user from answering the question everytime when add a new property. 
But not sure would it make sense for composed id, see https://loopback.io/doc/en/lb3/Model-definition-JSON-file.html#id-properties
IMO this is out of the scope of this PR though.

Reply:
I'll create a follow up issue to discuss composite IDs and how we will support them.